src/envvars.zig

```diff
@@ -0,0 +1,311 @@
+
+    // Type used to communicate between the Cache and the public interface of this env var.
+    const CachedInterface = union(enum) {
+        /// The environment variable hasn't been loaded yet.
+        undefined: void,
+        /// The environment variable has been loaded but its not set.
+        not_set: void,
+        /// The environment variable is set to a value.
+        value: NonOptionalReturnType,
+    };
+
+    const Cache = switch (T) {
+        .string => struct {
+            const PtrType = ?[*]const u8;
+            const LengthType = u64;
+
+            /// Indicates an environment variable hasn't been loaded yet.
+            const undefined_sentinel: LengthType = std.math.maxInt(LengthType);
+
+            /// Indicates an environment variable isn't set from the outside.
+            var value: std.atomic.Value(PtrType) = std.atomic.Value(PtrType).init(null);
+            var value_len: std.atomic.Value(u64) = .init(std.math.maxInt(LengthType));
+
+            /// Get the cached value.
+            pub inline fn getCached() CachedInterface {
+                const len = value_len.load(.monotonic);
+                if (len == undefined_sentinel) {
+                    return .{ .undefined = {} };
+                }
+
+                const v: PtrType = value.load(.monotonic);
+
+                return if (v) |ptr| .{ .value = ptr[0..len] } else .{ .not_set = {} };
+            }
+
+            pub inline fn reload(raw_env: ?[]const u8) ?NonOptionalReturnType {
+                if (raw_env) |ev| {
+                    value.store(ev.ptr, .monotonic);
+                    value_len.store(ev.len, .monotonic);
+                } else {
+                    value.store(null, .monotonic);
+                    value_len.store(0, .monotonic);
+                }
+
+                return raw_env;
+            }
+        },
+        .boolean => struct {
+            const ValueType = enum(u8) { undefined, not_set, no, yes };
+
+            var value = std.atomic.Value(ValueType).init(undefined);
+
+            pub inline fn getCached() CachedInterface {
+                return switch (value.load(.monotonic)) {
+                    .undefined => return .{ .undefined = {} },
+                    .not_set => return .{ .not_set = {} },
+                    .no => return .{ .value = false },
+                    .yes => return .{ .value = true },
+                };
+            }
+
+            pub inline fn reload(raw_env: ?[]const u8) ?NonOptionalReturnType {
+                if (raw_env == null) {
+                    value.store(.not_set, .monotonic);
+                    return null;
+                }
+
+                const true_values = .{
+                    "1",
+                    "true",
+                    "TRUE",
+                    "yes",
+                    "YES",
+                    "on",
+                    "ON",
+                };
```

Comment:
Usually I see environment variable truthiness handled the other way: all non-empty strings (except possibly `0`, `OFF`, etc.) are treated as on. Is this to match our existing behavior?